### PR TITLE
553 bug arjun bug fix

### PR DESCRIPTION
**Pull request by bradleybogg**
This PR addresses the following issues for Arjun Tool

1. Inconsistent component Name.
2. Lack of description in the tool tip.
3. Missing comments.
4. Introduce docs.
5. Grammar.
6. camelCase naming convention.
7. Docstring amendment.
8. Introduce runtime dependency checker

Please reach out if there are any issues.

### CODE REVIEW

**Review by michaeljpigott — APPROVED** [review]
Hi Bradley, 

This looks good. I just have a question about whether the constant description_userguide should be camel case, i.e. descriptionUserGuide. Obviously this is not a biggie and happy for the pull request to be merged with main. 

Please note that I was unable to test the checkAllCommandsAvailability function on my machine because I was unable to uninstall Arjun (despite best efforts). May need further application testing. 

Michael

File: src/components/Arjuntool/Arjuntool.tsx [hunk 1]
```diff
@@ -1,69 +1,114 @@
...
 import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
 
+// Component constants.
 const title = "Arjun";
+
+// Contains the description of the component.
 const description_userguide =
```
[the remark this answers is not in the document]
Comment (bradleybogg) [reply src/components/Arjuntool/Arjuntool.tsx:16]: This activity will be covered under baselining.

File: src/components/Arjuntool/Arjuntool.tsx [hunk 2]
```diff
@@ -78,44 +123,65 @@ export function Arjuntool() {
...
-                {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
-                <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
-            </Stack>
-        </form>
+        <>
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description_userguide}
```
[the remark this answers is not in the document]
Comment (bradleybogg) [reply src/components/Arjuntool/Arjuntool.tsx:170]: @michaeljpigott

Good pickup, however, this is using the syntax of the InstallationModal which should of had camelCase on its variables. To change it here is to change it's reference component. 
```suggestion
                    featureDescription={description_userguide}
```

[the remark this answers is not in the document]
Comment (bradleybogg) [reply src/components/Arjuntool/Arjuntool.tsx:170]: https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/issues/652
Raised to rectify

**Pull request merged**